src/dune/dpath.ml

```diff
@@ -27,6 +27,7 @@ let analyse_target (fn as original_fn) : target_kind =
             assert (String.length digest = 32);
```

Comment:
By the way, `analyse_target` has these annoying asserts which were pretty annoying last time I tried to use it for more things. I guess in your case it makes the error messages worse.